lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -110,6 +112,42 @@ def resolve_database_info_from_model(model_name)
         { error: e.full_message(highlight: false) }
       end
 
+      def resolve_association_target(params)
+        const = ActiveSupport::Inflector.safe_constantize(params[:model_name])
+        unless active_record_model?(const)
+          return {
+            result: nil,
+          }
+        end
+
+        association_klass = case params[:association_type].intern
+        when :has_many
+          ActiveRecord::Associations::Builder::HasMany.build(const, params[:association_name].intern, nil, {}).klass
+        when :belongs_to
+          ActiveRecord::Associations::Builder::BelongsTo.build(const, params[:association_name].intern, nil, {}).klass
+        when :has_one
+          ActiveRecord::Associations::Builder::HasOne.build(const, params[:association_name].intern, nil, {}).klass
+        when :has_and_belongs_to_many
+          ActiveRecord::Reflection::HasAndBelongsToManyReflection.new(params[:association_name], nil, {}, const).klass
+        else
+          return { error: "Unsupported association type #{params[:association_type]}" }
+        end
+
+        source_location = Object.const_source_location(association_klass.to_s)
+
+        {
+          result: {
+            location: source_location.first + ":" + source_location.second.to_s,
+          },
+        }
+      rescue NameError => e
+        {
+          result: {
+            error: e.message,
+          },
+        }
+      end
```

Comment:
I'm not sure the right behavior here; I opted to return the error message; but _I think_ would could return a nil result just like when the model name doesn't resolve properly. Thoughts?

Reply:
Yeah, I think `nil` is more appropriate here.

Thinks make me think though: Sometimes it's confusing for the user if they try something and the editor seems to do nothing. Perhaps there could be an `info` response, which prints something to the log in these kinds of situations.

Reply:
Switched it to `nil` for now. I like the idea of having info responses print to the log. Should we add an issue to add that support?